server/models/Group.js

```diff
@@ -223,6 +223,22 @@ module.exports = function(Sequelize, DataTypes) {
         });
       },
 
+      findOrAddUserWithRole(user, role) {
```

Comment:
if you say find or add user, then I expect this to return a User.
So you should add `include: [ {model: Sequelize.models.User } ]` and return `userGroup.User`

Where is this used?

I wonder if it wouldn't be simpler to just do `addUserWithRole` with the edge case of the user already being in the group for that role to simply return as if it wasn't already there.